internal/logging/logger.go

```diff
@@ -1,60 +1,61 @@
 package logging
 
 import (
-	"encoding/json"
-	"fmt"
 	"os"
 	"strings"
 
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
-// NewLogger creates a new default logger
-// it will need to be closed with
-// ```
-// defer logger.Desugar().Sync()
-// ```
-// to make sure all has been piped out before terminating
-func NewLogger(l string) *zap.SugaredLogger {
-	var lvl string
+type Logger struct {
+	*zap.SugaredLogger
+	security *SecurityLogger
```

Comment:
```suggestion
	*zap.SugaredLogger
	*SecurityLogger
```

if you were to use composition, wouldn't you be able to spare a lot of the methods defined in the Logger simply by exploiting the inheritance?

Reply:
You mean to reuse the same logger for security and system logs? If yes, then I would like to avoid this as we (probably) may have to log security logs to a different place. Also I wanted to make it create a distinction between the 2 types of logs.